code/_onclick/item_attack.dm

```diff
@@ -200,14 +200,12 @@
  * * params - Click params of this attack
  */
 /obj/item/proc/attack(mob/living/target_mob, mob/living/user, params)
-	var/signal_return = SEND_SIGNAL(src, COMSIG_ITEM_ATTACK, target_mob, user, params)
+	var/signal_return = SEND_SIGNAL(src, COMSIG_ITEM_ATTACK, target_mob, user, params) | SEND_SIGNAL(user, COMSIG_MOB_ITEM_ATTACK, target_mob, user, params)
 	if(signal_return & COMPONENT_CANCEL_ATTACK_CHAIN)
 		return TRUE
 	if(signal_return & COMPONENT_SKIP_ATTACK)
 		return FALSE
 
-	SEND_SIGNAL(user, COMSIG_MOB_ITEM_ATTACK, target_mob, user, params)
```

Comment:
Very few things: Ninja cloak, golem eating diamonds, gunpoint and stylecounter (and like uhh watcher? they cant wield items so I don't see an item override mattering much). 

The only buggy scenario that can occur is if someone attacks something with a weapon, the weapon blocks the attack and the mob also has a component that listens to attack. I guess if someone under gunpoint uses an item that refuses to attack they could still get shot? I think it's so incredibly edge case that it's probably fine

Reply:
Sounds good although it might still be worth passing it by @MrMelbert who has more knowledge of attack chain footguns than me.

Reply:
Seems fine

Reply:
Well if you wanted to avoid buggy behavior entirely you could do 
`
SEND_SIGNAL(src, COMSIG_ITEM_ATTACK, target_mob, user, params) || SEND_SIGNAL(user, COMSIG_MOB_ITEM_ATTACK, target_mob, user, params)
`
By using `||` instead of `|` the second signal is not sent if the first signal returns any value
And given any value = "attack fail", it would work perfect here

Reply:
Good idea!